packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -138,12 +160,20 @@ export class AssetStaging extends CoreConstruct {
 
   private readonly cacheKey: string;
 
+  private readonly sourceStats: fs.Stats;
+
   constructor(scope: Construct, id: string, props: AssetStagingProps) {
     super(scope, id);
 
     this.sourcePath = path.resolve(props.sourcePath);
     this.fingerprintOptions = props;
 
+    if (!fs.existsSync(this.sourcePath)) {
+      throw new Error(`Cannot find asset at ${this.sourcePath}`);
+    }
+
+    this.sourceStats = fs.statSync(this.sourcePath);
```

Comment:
can you explain why we need to store these stats? can't we just query them when needed?

Reply:
This is just to avoid "stating" the same path multiple times.